### PR TITLE
Handle system permissions + pass-through any permission attributes specified

### PR DESCRIPTION
**Pull request by jamesjhedley**
Fixes blackberry/BB10-WebWorks-Packager#194

### CODE REVIEW

**Comment by jamesjhedley** [comment]
FYI @mdandrea .

**Comment by mdandrea** [comment]
Great. :)

**Comment by jamesjhedley** [comment]
@rwmtse Code review? :)

File: test/unit/lib/config-parser.js [hunk 1]
```diff
@@ -161,18 +161,6 @@ describe("config parser", function () {
         expect(fileManager.cleanSource).toHaveBeenCalled();
     });
 
-    it("adds the access_internet permission if unprovided", function () {
```
Comment (rwmtse) [inline test/unit/lib/config-parser.js:171]: Do you need to add new unit test cases for parsing <rim:permission system="true">xyz_permission/rim:permission? And there should also be a test for parsing rim:permission which does not have the system attribute?

**Comment by nukulb** [comment]
@jkeshavarzi - if this permission that is need for push now, you need update the test-app along with this pull request please.

**Comment by rwmtse** [comment]
r+ from me

**Comment by tracyli** [comment]
tested with http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-system-permissions/6/: smoke test got passed and push api works fine.

By the way, we only need to set permission as rim:permit_sys_use_consumer_push/rim:permit to make push work; there's no difference between adding an attribute 'system' to be 'true' or 'false', either could make push work.

**Comment by mdandrea** [comment]
There is a difference in the manifest between the two. And signing might not work if you don't use the system one because of that.

From: Tracy Li
To: blackberry-webworks/BB10-Webworks-Packager
To: Matthew D'Andrea
Reply To: blackberry-webworks/BB10-Webworks-Packager
Re: [BB10-Webworks-Packager] Handle system permissions + pass-through any permission attributes specified (#92)
2012-10-11 11:20:21 PM

tested with http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-system-permissions/6/: smoke test got passed and push api works fine.

By the way, we only need to set permission as rim:permit_sys_use_consumer_push/rim:permit to make push work; there's no difference between adding an attribute 'system' to be 'true' or 'false', either could make push work.

\u2014
Reply to this email directly or view it on GitHubhttps://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/92#issuecomment-9361733.

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

**Comment by tracyli** [comment]
retested with packager http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-system-permissions/6/:  push works fine with signed or unsigned.

**Comment by tracyli** [comment]
tested in windows and mac with packager http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-system-permissions/8/: smoke test passed; push works with sign and unsigned; _sys_use_consumer_push shows up at section Entry-Point-User-Actions when 'system' attribute is set as false and error would occur when signing; sys_use_consumer_push shows up at section Entry-Point-System-Actions when 'system' attribute is set as true and signing works without problem.

one question: if we have other permissions such as ' read_device_identifying_information', there's permission information showing up at the beginning: 'Device Identification Information Allows this app to access device id such as serial num & pin' item has been checked
There's no such kind of permission info for system permission even it has been set as 'true' - is it what expected?

**Comment by nukulb** [comment]
@tracli - that is fine, some permissions are of type prompt, other are not.
we have no control over which permission pops up and which doesn't, its controlled by the OS

**Pull request closed without merging**